### PR TITLE
Turn deprecation message analyze tests back on

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Additionally:
- Deprecation message analyzer: use AST more instead of regex 
- Deprecation message analyzer: removes some tests since they are in the formatter's jurisdiction now
- update the analyzer test fixture to fix a couple line numbers
- `@_debugOnly` set `multiline` back to false. The source we get back from the `ASTNode` doesn't have any line feed characters.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by goderbauer** [comment]
@LongCatIsLooong looks like some line numbers are off in tests:

```
00:10 +67 ~3 -1: test/analyze_test.dart: analyze.dart - clampDouble [E]
  Expected: '╔═╡ERROR #1╞════════════════════════════════════════════════════════════════════\n'
              '║ packages/flutter/lib/bar.dart:37: input.clamp(0.0, 2)\n'
              '║ packages/flutter/lib/bar.dart:38: input.toDouble().clamp(0, 2)\n'
              '║ packages/flutter/lib/bar.dart:42: nullableInt?.clamp(0, 2.0)\n'
              '║ packages/flutter/lib/bar.dart:43: nullableDouble?.clamp(0, 2)\n'
              '║ packages/flutter/lib/bar.dart:48: nullableInt?.clamp\n'
              '║ packages/flutter/lib/bar.dart:50: nullableDouble?.clamp\n'
              '║ \n'
              '║ For performance reasons, we use a custom "clampDouble" function instead of using "double.clamp".\n'
              '╚═══════════════════════════════════════════════════════════════════════════════\n'
              ''
    Actual: '╔═╡ERROR #1╞════════════════════════════════════════════════════════════════════\n'
              '║ packages/flutter/lib/bar.dart:44: input.clamp(0.0, 2)\n'
              '║ packages/flutter/lib/bar.dart:45: input.toDouble().clamp(0, 2)\n'
              '║ packages/flutter/lib/bar.dart:49: nullableInt?.clamp(0, 2.0)\n'
              '║ packages/flutter/lib/bar.dart:50: nullableDouble?.clamp(0, 2)\n'
              '║ packages/flutter/lib/bar.dart:55: nullableInt?.clamp\n'
              '║ packages/flutter/lib/bar.dart:57: nullableDouble?.clamp\n'
              '║ \n'
              '║ For performance reasons, we use a custom "clampDouble" function instead of using "double.clamp".\n'
              '╚═══════════════════════════════════════════════════════════════════════════════\n'
              ''
```

```
00:10 +68 ~3 -2: test/analyze_test.dart: analyze.dart - verifyDeprecations [E]
  Expected: '╔═╡ERROR #1╞════════════════════════════════════════════════════════════════════\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:12: Deprecation notice should be a grammatically correct sentence and start with a capital letter; see style guide: [https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md\n](https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md%5cn)'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:18: Deprecation notice should be a grammatically correct sentence and end with a period; notice appears to be "Also bad grammar".\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:23: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:26: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:31: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:50: Deprecation notice does not accurately indicate a beta branch version number; please see https://flutter.dev/docs/development/tools/sdk/releases to find the latest beta build version number.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:56: Deprecation notice does not accurately indicate a beta branch version number; please see https://flutter.dev/docs/development/tools/sdk/releases to find the latest beta build version number.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:79: Deprecation notice does not match required pattern. You might have used double quotes (") for the string instead of single quotes (\').\n'
              '║ See: [https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes\n](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes%5cn)'
              '╚═══════════════════════════════════════════════════════════════════════════════\n'
              ''
    Actual: '╔═╡ERROR #1╞════════════════════════════════════════════════════════════════════\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:14: Deprecation notice should be a grammatically correct sentence and start with a capital letter; see style guide: [https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md\n](https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md%5cn)'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:20: Deprecation notice should be a grammatically correct sentence and end with a period; notice appears to be "Also bad grammar".\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:25: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:28: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:33: Deprecation notice must be an adjacent string.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:52: Deprecation notice does not accurately indicate a beta branch version number; please see https://flutter.dev/docs/development/tools/sdk/releases to find the latest beta build version number.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:58: Deprecation notice does not accurately indicate a beta branch version number; please see https://flutter.dev/docs/development/tools/sdk/releases to find the latest beta build version number.\n'
              '║ test/analyze-test-input/root/packages/foo/deprecation.dart:81: Deprecation notice does not match required pattern. You might have used double quotes (") for the string instead of single quotes (\').\n'
              '║ See: [https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes\n](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes%5cn)'
              '╚═══════════════════════════════════════════════════════════════════════════════\n'
              ''
```

from https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8727802449131124289/+/u/run_test.dart_for_framework_tests_shard_and_subshard_misc/stdout

File: dev/bots/analyze.dart [hunk 1]
```diff
@@ -2557,29 +2549,21 @@ const Set<String> kExecutableAllowlist = <String>{
   'bin/flutter-dev',
   'bin/internal/update_dart_sdk.sh',
   'bin/internal/update_engine_version.sh',
-
```
Comment (goderbauer) [inline dev/bots/analyze.dart:2560]: I believe these blank lines are here to group the strings. Any good reason to remove them?

File: dev/bots/test/analyze-test-input/root/packages/flutter/lib/bar.dart [hunk 2]
```diff
@@ -15,6 +15,13 @@ class Foo {
 
   @_debugOnly
   final Map<String, String>? bar = kDebugMode ? null : <String, String>{};
+
+  // dart format off
```
Comment (goderbauer) [inline dev/bots/test/analyze-test-input/root/packages/flutter/lib/bar.dart:19]: Maybe leave a comment here explaining why format is turned off (e.g. to check that the annotation also work properly if the annotated things goes over multiple lines)?

File: dev/bots/analyze.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply dev/bots/analyze.dart]: I don't know why my IDE decides to remove them. Restored.

File: dev/bots/test/analyze-test-input/root/packages/flutter/lib/bar.dart [hunk 3]
```diff
@@ -16,11 +16,9 @@ class Foo {
   @_debugOnly
   final Map<String, String>? bar = kDebugMode ? null : <String, String>{};
 
-  // dart format off
+  // dart format off, checks the annotation works for multiline expressions
```
Comment (goderbauer) [inline dev/bots/test/analyze-test-input/root/packages/flutter/lib/bar.dart]: You may have to put this comment on the line above (looks like dartfmt was applied to the next line..)

File: dev/bots/test/analyze_test.dart [hunk 4]
```diff
@@ -54,51 +54,42 @@ void main() {
...
+      () => verifyDeprecations(testRootPath, minimumMatches: 2),
+      shouldHaveErrors: true,
+    );
+    final String lines = <String>[
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:14: Deprecation notice should be a grammatically correct sentence and start with a capital letter; see style guide: STYLE_GUIDE_URL',
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:20: Deprecation notice should be a grammatically correct sentence and end with a period; notice appears to be "Also bad grammar".',
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:25: Deprecation notice must be an adjacent string.',
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:28: Deprecation notice must be an adjacent string.',
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:33: Deprecation notice must be an adjacent string.',
+          '║ test/analyze-test-input/root/packages/foo/deprecation.dart:52: Deprecation notice does not accurately indicate a beta branch version number; please see RELEASES_URL to find the latest beta build version number.',
```
Comment (goderbauer) [inline dev/bots/test/analyze_test.dart:68]: Surprised that `test8` (line 44) doesn't produce a warning here. According to `analyze-test-input/root/packages/foo
/deprecation.dart` it "should fail".

**Review by goderbauer — COMMENTED** [review]
Looks like one file is missformated: 

```
Found 1 Dart file which was formatted incorrectly.
To fix, run `dart format dev/bots/utils.dart`
```

https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8727398808005744049/+/u/run_test.dart_for_analyze_shard_and_subshard_None/stdout

File: dev/bots/test/analyze_test.dart (hunk 4 above)
Comment (LongCatIsLooong) [reply dev/bots/test/analyze_test.dart:68]: I check this on an older commit that is a couple month old:

https://github.com/flutter/flutter/blob/d755bc222b2861146d64cf0570c4ba37db970918/dev/bots/test/analyze_test.dart
https://github.com/flutter/flutter/blob/d755bc222b2861146d64cf0570c4ba37db970918/dev/bots/test/analyze-test-input/root/packages/foo/deprecation.dart

Comment (LongCatIsLooong) [reply dev/bots/test/analyze_test.dart:68]: From https://github.com/flutter/flutter/blob/d755bc222b2861146d64cf0570c4ba37db970918/dev/bots/analyze.dart#L635-L639

From the code I think the test case is just mislabeled and it should succeed?

Comment (goderbauer) [reply dev/bots/test/analyze_test.dart:68]: I see. Thanks for checking. Can you correct the label of the test case to indicate that it should pass?

**Review by goderbauer — APPROVED** [review]
LGTM

Thank you for fixing this up!

**Pull request merged**